aws/resource_aws_lb_listener.go

```diff
@@ -493,6 +493,14 @@ func resourceAwsLbListenerCreate(d *schema.ResourceData, meta interface{}) error
 		return nil
 	})
 
+	if isResourceTimeoutError(err) {
```

Comment:
When the error is a `resource.TimeoutError` and we hit this condition, its important that we reset the error object so the later error checking logic is not triggered. Otherwise as written this will work like the following:

1. `resource.Retry` sets top level `err` to `resource.TimeoutError`
1. `isResourceTimeoutError(err)` conditional runs `CreateListener` again (great!)
1. If `CreateListener` returns an error, it returns the error
1. If `CreateListener` doesn't return an error, the `isResourceTimeoutError()` conditional exits, but the below error check (`if err != nil`) will still have the original `resource.TimeoutError` error, so it will still return an error

The trick here is using a direct `=` assignment in Go instead of `:=`

This allows us to simplify this logic like the following:

```go
	// ... new logic here ...
	if isResourceTimeoutError(err) {
		_, err = elbconn.CreateListener(params)
	}

	// ... existing error handling below here ...
	if err != nil {
		return fmt.Errorf("Error creating LB Listener: %s", err)
	}

	// ... existing happy path ...
```

You'll now get the outcome you're expecting both ways when `resource.Retry` has a timeout error:

1. `resource.Retry` sets top level `err` to `resource.TimeoutError`
1. `isResourceTimeoutError(err)` conditional runs `CreateListener` again (great!)
1. If `CreateListener` returns an error, it resets the top level error object and the below error check (`if err != nil`) will return _that_ error
1. If `CreateListener` doesn't return an error, the top level error object is reset to `nil` and the below error check (`if err != nil`) will not trigger so the resource will continue on its merry way on the happy path

Hope this helps!